go.mod

```diff
@@ -9,31 +9,31 @@ require (
 	github.com/go-git/go-git/v5 v5.4.2
 	github.com/go-logr/logr v0.4.0
 	github.com/go-logr/zapr v0.4.0
-	github.com/golang/mock v1.5.0
+	github.com/golang/mock v1.6.0
 	github.com/google/go-github/v35 v35.2.0
-	github.com/mrajashree/etcdadm-controller v0.1.3
-	github.com/onsi/gomega v1.15.0
+	github.com/mrajashree/etcdadm-controller v1.0.0-rc3
+	github.com/onsi/gomega v1.16.0
 	github.com/pkg/errors v0.9.1
-	github.com/spf13/cobra v1.1.3
+	github.com/spf13/cobra v1.2.1
 	github.com/spf13/pflag v1.0.5
-	github.com/spf13/viper v1.7.1
+	github.com/spf13/viper v1.9.0
 	github.com/stretchr/testify v1.7.0
 	go.uber.org/zap v1.19.0
-	golang.org/x/crypto v0.0.0-20210513164829-c07d793c2f9a
-	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
-	golang.org/x/sys v0.0.0-20210817190340-bfb29a6856f2
+	golang.org/x/crypto v0.0.0-20210817164053-32db794688a5
+	golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f
+	golang.org/x/sys v0.0.0-20210823070655-63515b42dcdf
 	gopkg.in/square/go-jose.v2 v2.6.0
 	k8s.io/api v0.22.2
 	k8s.io/apimachinery v0.22.2
 	k8s.io/client-go v0.22.2
-	sigs.k8s.io/cluster-api v0.3.11-0.20210430210359-402a4524f006
-	sigs.k8s.io/cluster-api-provider-vsphere v0.7.8
+	sigs.k8s.io/cluster-api v1.0.2
+	sigs.k8s.io/cluster-api-provider-vsphere v1.0.1
 	sigs.k8s.io/controller-runtime v0.10.3
 	sigs.k8s.io/yaml v1.3.0
 )
 
-// exclude un-required transitive dependency from cluster-api-provider-vsphere v0.7.8
-exclude sigs.k8s.io/cluster-api v0.3.14
+// exclude un-required transitive dependency from cluster-api-provider-vsphere v1.0.1
+exclude sigs.k8s.io/cluster-api v0.0.0-00010101000000-000000000000
```

Comment:
Why is this not `v1.0.0` as it shows in upstream?

Reply:
otherwise `go mod tidy` throws this error
```
go: sigs.k8s.io/cluster-api-provider-vsphere@v1.0.1 requires
	sigs.k8s.io/cluster-api/test@v1.0.0 requires
	sigs.k8s.io/cluster-api@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

Reply:
So I tried to checkout this change and remove this line and add a replace similar to underneath like how it is done here, and then `go mod tidy` succeeds. https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/v1.0.1/go.mod#L37

I am not sure what is different, but plan on taking a look more into it tomorrow. Wondering if that is better than excluding a dependency? But may not be that different tbh, just looking at what upstream does.

Reply:
I too tried this out locally with capi tag v1.0.2 `sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.0.2` under replace and go mod tidy worked